test/integration/v29_int_test.go

```diff
@@ -0,0 +1,174 @@
+package integration
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/ActiveState/cli/internal/constants"
+	"github.com/ActiveState/cli/internal/environment"
+	"github.com/ActiveState/cli/internal/testhelpers/e2e"
+	"github.com/ActiveState/cli/internal/testhelpers/tagsuite"
+	"github.com/stretchr/testify/suite"
+)
+
+type V29TestSuite struct {
+	tagsuite.Suite
+}
+
+type versionData struct {
+	Version string `json:"version"`
+}
+
+const rcVersion = "0.28.7-SHAf566db1"
+const rcTag = "v29-update"
+
+func (suite *V29TestSuite) installReleaseCandidate(ts *e2e.Session) string {
+	scriptExt := ".sh"
+	shell := "bash"
+	extraEnv := []string{}
+	if runtime.GOOS == "windows" {
+		scriptExt = ".ps1"
+		shell = "powershell.exe"
+		extraEnv = []string{"SHELL="}
+	}
+
+	script := filepath.Join(environment.GetRootPathUnsafe(), "installers", "install"+scriptExt)
+
+	cp := ts.SpawnCmdWithOpts(
+		shell,
+		e2e.WithArgs(script, "-t", ts.Dirs.Work, "-b", "beta"),
+		e2e.AppendEnv(
+			fmt.Sprintf("%s=%s", constants.OverwriteDefaultInstallationPathEnvVarName, filepath.Join(ts.Dirs.Work, "multi-file")),
+		),
+		e2e.AppendEnv(extraEnv...),
+	)
+
+	cp.Expect("proceed with install?")
+	cp.SendLine("Y")
+	cp.Expect(fmt.Sprintf("Fetching the latest version: %s", rcVersion))
+	if runtime.GOOS == "windows" {
+		cp.Expect("successfully installed to", 30*time.Second)
+	} else {
+		cp.Expect("Installing to", 30*time.Second)
+		cp.Expect("Allow $PATH to be appended in your")
+		cp.SendLine("n")
+		cp.Expect("State Tool installation complete")
+	}
+	cp.ExpectExitCode(0)
+
+	stateExe := filepath.Join(ts.Dirs.Work, "state")
+	if runtime.GOOS == "windows" {
+		stateExe = stateExe + ".exe"
+	}
+	suite.compareVersionedInstall(ts, stateExe, rcVersion, suite.Equal)
+
+	return stateExe
+}
+
+// TestTaggedUpdateFlow is meant to test whether installing our release-candidate will update correctly
+// It is supposed to not attempt an auto-update, and the tag should be set
+func (suite *V29TestSuite) TestTaggedUpdateFlow() {
+	suite.OnlyRunForTags(tagsuite.V29Update, tagsuite.Critical)
+
+	ts := e2e.New(suite.T(), true)
+	defer ts.Close()
+
+	stateExe := suite.installReleaseCandidate(ts)
+
+	// ensure tag is in config yaml file
+	f, err := os.ReadFile(filepath.Join(ts.Dirs.Config, "config.yaml"))
+	suite.Require().NoError(err)
+	suite.Require().Contains(string(f), fmt.Sprintf(`tag: %s`, rcTag))
+
+	suite.Run("Tagged RC should not update", func() {
+		cp := ts.SpawnCmdWithOpts(
+			stateExe,
+			e2e.WithArgs("update"),
+			e2e.AppendEnv(
+				fmt.Sprintf("%s=%s", constants.OverwriteDefaultInstallationPathEnvVarName, filepath.Join(ts.Dirs.Work, "multi-file")),
+			),
+		)
+		cp.Expect("You are already using the latest State Tool version")
+		cp.ExpectExitCode(0)
+	})
+
+	suite.Run("Update without tag", func() {
+		// remove update-tag
+		err := os.Remove(filepath.Join(ts.Dirs.Config, "config.yaml"))
```

Comment:
Rather than removing the config file entirely, if we just removed the tag entry we could also capture the `importLegacyConfig` part of the flow.  Though that is admittedly more work.

Reply:
:-) That's what I figured.